test/integration/src/test/java/io/pravega/test/integration/LargeEventTest.java

```diff
@@ -0,0 +1,523 @@
+import io.pravega.client.stream.impl.ClientFactoryImpl;
+import io.pravega.common.concurrent.ExecutorServiceHelpers;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.segmentstore.contracts.StreamSegmentStore;
+import io.pravega.segmentstore.contracts.tables.TableStore;
+import io.pravega.segmentstore.server.host.handler.PravegaConnectionListener;
+import io.pravega.segmentstore.server.store.ServiceBuilder;
+import io.pravega.segmentstore.server.store.ServiceBuilderConfig;
+import io.pravega.shared.protocol.netty.PravegaNodeUri;
+import io.pravega.shared.protocol.netty.ReplyProcessor;
+import io.pravega.test.common.LeakDetectorTestSuite;
+import io.pravega.test.common.TestUtils;
+import io.pravega.test.common.TestingServerStarter;
+import io.pravega.test.integration.demo.ControllerWrapper;
+import lombok.Cleanup;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang3.RandomUtils;
+import org.apache.curator.test.TestingServer;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+public class LargeEventTest extends LeakDetectorTestSuite {
+
+    private static final int NUM_WRITERS = 2;
+    private static final int NUM_READERS = 1;
+    private static final String SCOPE_NAME = "scope";
+    private final int servicePort = TestUtils.getAvailableListenPort();
+    private final int controllerPort = TestUtils.getAvailableListenPort();
+
+
+    private TableStore tableStore;
+    private StreamSegmentStore store;
+    private AtomicLong eventReadCount;
+    private AtomicLong eventWriteCount;
+    private AtomicBoolean stopReadFlag;
+    private ServiceBuilder serviceBuilder;
+    private TestingServer zkTestServer = null;
+    private PravegaConnectionListener server = null;
+    private ControllerWrapper controllerWrapper = null;
+    private Controller controller = null;
+    private ScheduledExecutorService writerPool;
+    private ScheduledExecutorService readerPool;
+    private ConcurrentLinkedQueue<ByteBuffer> eventsReadFromPravega;
+    private ConcurrentHashMap<Integer, ByteBuffer> eventsWrittenToPravega;
+
+
+
+    @Before
+    public void setup() throws Exception {
+        resetReadWriteObjects();
+        String serviceHost = "localhost";
+        int containerCount = 4;
```

Comment:
I think 1 container should do the trick for this test. I don't see a need for 4 (that would just make it slower).